ext/phar/tar.c

```diff
@@ -488,7 +488,14 @@ int phar_parse_tarfile(php_stream* fp, char *fname, int fname_len, char *alias,
 			entry.link = estrdup(hdr->linkname);
 		}
 		phar_set_inode(&entry);
-		if ((newentry = zend_hash_str_add_mem(&myphar->manifest, entry.filename, entry.filename_len, (void*)&entry, sizeof(phar_entry_info))) == NULL) {
+
+		if (zend_hash_str_exists(&myphar->manifest, entry.filename, entry.filename_len)) {
+			newentry = zend_hash_str_update_mem(&myphar->manifest, entry.filename, entry.filename_len, (void*)&entry, sizeof(phar_entry_info));
+		} else {
+			newentry = zend_hash_str_add_mem(&myphar->manifest, entry.filename, entry.filename_len, (void*)&entry, sizeof(phar_entry_info));
+		}
```

Comment:
It should be sufficient to use only an `zend_hash_str_update_mem` call here. In that case you should also replace the `newentry == NULL` check below with a `ZEND_ASSERT(newentry != NULL)`, as it can no longer occur.